### PR TITLE
Add support for partial syntax

### PR DESCRIPTION
**Pull request by ankrgyl**
This enables you to run, for example, `ClosedQA.partial({criteria: "evaluate the sql query"})` without having to create a wrapper function.

### CODE REVIEW

File: js/partial.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Scorer, ScorerArgs } from "@braintrust/core";
+
+export interface ScorerWithPartial<Output, Extra>
+  extends Scorer<Output, Extra> {
+  partial: <T extends Partial<ScorerArgs<Output, Extra>>>(
```
Comment (manugoyal) [inline js/partial.ts:5]: What do you think about something like this?

```
partial: <T extends keyof Extra>(args: { [K in T]: Extra[K] }) => Scorer<Output, Omit<Extra, T> & Partial<Pick<Extra, T>>>
```

Here, `args` only permits sub-members of `Extra` which match exactly the type signature in `Extra`. So we cannot provide arguments which don't exist in `Extra`, nor can we provide `undefined` for arguments which are required in `Extra`, and thus launder an `undefined` into the actually-invoked function.

Also, this feels like a nice general utility. Something like:
```
function partial<F extends (args: any) => any, PartialArgs extends keyof Parameters<F>[0]>(
    f: F,
    partialArgs: { [K in PartialArgs]: Parameters<F>[0][K] }
) {
    function ret(args: Omit<Parameters<F>[0], PartialArgs> & Partial<Pick<Parameters<F>[0], PartialArgs>>) {
        return f({
            ...partialArgs,
            ...args,
        });
    }
    // Do any function metadata adjustment too.
    return ret;
}
```

File: py/autoevals/partial.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    kwargs["expected"] = expected
+                return self._run_eval_sync(output, **{**partial_kwargs, **kwargs})
+
+            @classmethod
+            def _partial_args(cls):
+                return {**partial_kwargs}
+
+            @classmethod
+            def _cls_name(cls_):
+                return cls._cls_name()
```
Comment (manugoyal) [inline py/autoevals/partial.py:24]: Why not use the standard `cls_.__name__`?

**Review by manugoyal — COMMENTED** [review]
generally lgtm, though some thoughts about the signature/implementation of `partial`.

File: py/autoevals/partial.py (hunk 2 above)
Comment (ankrgyl) [reply py/autoevals/partial.py:24]: Good catch can remove this now

File: js/partial.ts (hunk 1 above)
Comment (ankrgyl) [reply js/partial.ts:5]: Nice. The new type signature works for me. I'll leave the generic utility function out for now.

**Review by manugoyal — APPROVED** [review]

**Pull request merged**